### DISCUSSION TITLE
Assign Midi CC Values To Kit Rows

### DISCUSSION THREAD

**Discussion by DamianGolf666** [SynthstromAudible/DelugeFirmware#2161] [Ideas]
Apologies if this has been brought up (I have attempted a search). I think it would be a worthwhile addition to be able to assign CC values to kit independant rows. I have multiple external drum machines that I have built Deluge kit templates for, but the Deluge is limiting the ability to adjust CC parameters for these, ie I cannot specifically set the pitch value for a kit, tom or snare.

**Comment by Ponchomouse** [comment]
Hi, came here to say this it is such a good idea. I have multiple external synths and a book of printed cc implementation charts so I can automate say cutoff on something. The deluge is amazing at this as you can just have a clip that is essentially a custom lfo controlling external cc. It would be great if we could name the ccs or better yet save into kits so they are mapped when we open it.

**Reply by seangoodvibes** [reply]
> Hi, came here to say this it is such a good idea. I have multiple external synths and a book of printed cc implementation charts so I can automate say cutoff on something. The deluge is amazing at this as you can just have a clip that is essentially a custom lfo controlling external cc. It would be great if we could name the ccs or better yet save into kits so they are mapped when we open it.

you can name the cc's in the nightly build. 

https://github.com/SynthstromAudible/DelugeFirmware/blob/community/docs/features/midi_device_definition_files.md

**Reply by Ponchomouse** [reply]
Ah great, man I love how much cool stuff is coming to the Deluge. Great work team.

**Comment by DamianGolf666** [comment]
Ah great, but just to clarify: can you assign CC's to kit rows on the Deluge? That's what I'm missing, I'd be able to get by without naming the CC's, but it's that I cannot for instance create a kit that is mapped to an external drum machine, then assign a pitch change CC to the snare note row. Love all yr work Sean!

**Reply by seangoodvibes** [reply]
Nah you're not wrong you can't send cc's from a midi kit row. I don't think anyone's working on this at the moment.

**Reply by Ponchomouse** [reply]
Ah with you, I have the tr8s mapped as a kit row but use a midi clip for all automation. Nice as you can have different clip lengths and different automation clips but yeah would be nice if this was built into the kit rather than separate.

**Comment by DamianGolf666** [comment]
Ah true, bugger - well same idea stands :) 
I have several external drum machines/samplers that are XOX style programmable machines, but there's just no way to have control over all their CC functions because of the Deluge kit + CC limitation. And yeah the midi clip for automation can be problematic I find when you have clips with 'A' versions. All sorts of launch mischief :)

**Reply by RJ-Eckie** [reply]
I'm with you here. There's a lot on my wishlist for better interaction with external gear, as that's my main use-case. I do want to start focusing on developing midi-stuff in the somewhat-near future, but no promises as I want to focus on making the code easier to navigate first.
